tests/pubsub/testscoring.nim

```diff
@@ -42,3 +38,441 @@ suite "GossipSub Scoring":
       gossipSub.gossipsub.peers(topic) == 0
       # also ensure we cleanup properly the peersInIP table
       gossipSub.peersInIP.len == 0
+
+  asyncTest "Time in mesh scoring (P1)":
+    const topic = "foobar"
+    var (gossipSub, conns, peers) =
+      setupGossipSubWithPeers(3, topic, populateMesh = true)
+    defer:
+      await teardownGossipSub(gossipSub, conns)
```

Comment:
This should be on a suite-wise `teardown` block, since all tests use it. Also, consider making `gossipsub`, `conns` and `peers` suite-wise `{.threadvar.`}